dvc/tree/s3.py

```diff
@@ -353,6 +353,13 @@ def get_file_hash(self, path_info):
                 size=obj.content_length,
             )
 
+    def _upload_fobj(self, fobj, to_info):
+        from boto3.s3.transfer import TransferConfig
+
+        config = TransferConfig(max_concurrency=1, use_threads=False)
```

Comment:
same question as for azure

Reply:
@isidentical And one more question: when we don't limit the concurency here, are there any specific requirements for the fobj that it is uploading from? E.g. if there are multiple threads, do they just `read()` sequentially or do they try to do some seek() magic?

Reply:
> E.g. if there are multiple threads, do they just read() sequentially or do they try to do some seek() magic?

The interface that `to_remote` passes to them doesn't implement the full file-like protocol (no seek etc), so in theory they are not able to do that.

Reply:
So you've tried it and it didn't break, right?